src/shared/utils/extractUploads.js

```diff
@@ -1,86 +0,0 @@
-import countBy from 'lodash/countBy'
-import groupBy from 'lodash/groupBy'
-
-import { UploadStateEnum, UploadTypeEnum } from 'shared/utils/commit'
-
-function humanReadableOverview(state, count) {
-  if (state === UploadStateEnum.error) return 'errored'
-  if (state === UploadStateEnum.processed) return 'successful'
-  if (state === UploadStateEnum.complete) return 'carried forward'
-  if (state === UploadStateEnum.started) return 'started'
-}
-
-function deleteDuplicateCFFUploads({ uploads }) {
-  const nonCFFFlags = []
-  const duplicateUploads = uploads && [...uploads]
-
-  // Get all the flags from "uploaded" uploads
-  duplicateUploads?.forEach((upload) => {
-    if (
-      upload?.uploadType !== UploadTypeEnum.CARRIED_FORWARD &&
-      upload?.flags
-    ) {
-      nonCFFFlags.push(...upload.flags)
-    }
-  })
-
-  // Filter out cff uploads with repeated flags
-  // We're looping in reverse as splicing reindexes the array, making us
-  // skip the index that's removed. Indices are preserved when you
-  // loop backwards.
-  for (let index = duplicateUploads.length - 1; index >= 0; index--) {
```

Comment:
Now we're going through all uploads AND all CFF for each upload with a "break" if we find one early. In the worst case this is a quadratic function n*m where n is the number of uploads and m is the number of CFF we have

splice is also an _incredibly_ expensive operation because we are creating a new list each time we call that function.